store/src/java/com/zimbra/cs/service/formatter/LdifFormatter.java

```diff
@@ -86,77 +105,211 @@ public void formatCallback(UserServletContext context) throws UserServletExcepti
-                boolean valueBase64Encoded = false;
-                if (ContactConstants.A_userCertificate.equals(field)
-                    || ContactConstants.A_userSMIMECertificate.equals(field)) {
-                    // base64 encoded attribute appended with double colon
-                    sb.append(":: ");
-                    sb.append(value);
-                    sb.append(NEW_LINE);
-                    return;
+                String encodedValue = encode ? encodeValue(value) : value;
+                if (includeAll) {
+                    for (String ldapAttr : ldapAttrs) {
+                        constructLDIFEntry(ldapAttr, encodedValue, sb, !encodedValue.equals(value));
+                    }
+                } else {
+                    String ldapAttr = ldapAttrs[ldapAttrs.length - 1];
+                    constructLDIFEntry(ldapAttr, encodedValue, sb, !encodedValue.equals(value));
                 }
-                if (value.chars().allMatch(c -> c < 128)) {
-                    boolean containsCharsOtherThanSAFE_CHAR = false;
-                    boolean beginsWithCharOtheThanSAFE_INIT_CHAR = false;
-                    boolean isFirstChar = true;
-                    char[] chars = value.toCharArray();
-                    char lastChar = chars[chars.length - 1];
-                    if (lastChar != SPACE) {
-                        for (int i = 0; i < chars.length; i++) {
-                            char c = chars[i];
-                            if (isFirstChar) {
-                                if (c == NUL || c == LF || c == CR || c == SPACE || c == COLON
-                                    || c == LESSTHAN) {
-                                    beginsWithCharOtheThanSAFE_INIT_CHAR = true;
-                                    break;
-                                }
-                                isFirstChar = false;
-                            }
-                            if (c == NUL || c == LF || c == CR) {
-                                containsCharsOtherThanSAFE_CHAR = true;
-                                break;
-                            }
-                        }
-                        // if value contains characters other than those defined
-                        // as SAFE-CHAR, or begins with a character other than
-                        // those defined as SAFE-INIT-CHAR, MUST be base-64
-                        // encoded.
-                        if (containsCharsOtherThanSAFE_CHAR
-                            || beginsWithCharOtheThanSAFE_INIT_CHAR) {
-                            value = new String(Base64.encodeBase64(value.getBytes()));
-                            valueBase64Encoded = true;
+            }
+            return true;
+        }
+        return false;
+    }
+
+    private static void constructLDIFEntry(String name, String value, StringBuilder sb, boolean isEncoded) {
+        sb.append(name);
+        String separator = isEncoded ? ":: " : ": ";
```

Comment:
Perhaps add back in the comment `// base64 encoded attribute appended with double colon`

Reply:
Done